internal/incident/incidents.go

```diff
@@ -77,8 +77,14 @@ func GetCurrent(
 				return nil, false, errors.New("failed to load incident rule escalation states")
 			}
 
+			incident.runtimeConfig.RLock()
+			defer incident.runtimeConfig.RUnlock()
+
 			for _, state := range states {
 				incident.EscalationState[state.RuleEscalationID] = state
+
+				escalation := incident.runtimeConfig.GetRuleEscalation(state.RuleEscalationID)
+				incident.Rules[escalation.RuleID] = struct{}{}
 			}
```

Comment:
`defer` is executed only when leaving the function. Until that happens, this function does quite a few things that don't need this lock including database queries.

Now it's a bit annoying that `defer` only works on function scopes, so maybe put this block of code into it's own function, something like `func (i *Incident) RestoreState(states []*EscalationState)`.